plugins/anchor-links/index.js

```diff
@@ -56,18 +56,25 @@ function processHeading(node, compatibilitySlug, links, headings) {
     .substring(level + 1)
     .replace(/<\/?[^>]*>/g, '') // Strip html
     .replace(/\(\(#.*?\)\)/g, '') // Strip anchor link aliases
+    .replace(/»/g, '') // Safeguard against double-running this plugin
     .replace(/\s+/g, ' ') // Collapse whitespace
     .trim()
 
-  // generate the slug and add a target element to the headline
+  // generate the slug and use it as the headline's id property
   const slug = generateSlug(text, links)
-  node.children.unshift({
-    type: 'html',
-    value: `<a class="__target-h" id="${slug}" aria-hidden="true"></a>`,
-  })
+  node.data = {
+    ...node.data,
+    hProperties: { ...node.data?.hProperties, id: slug },
+  }
```

Comment:
This is the core change: rather than injecting a new `<a />` tag with the desired `id`, we add the `id` directly to the heading `node`.